### DISCUSSION TITLE
No History Nor Favourites

### DISCUSSION THREAD

**Discussion by nevdelap** [rustdesk/rustdesk#10206] [General]
### Bug Description

I see #9119, which looks like the same thing but is closed/fixed. That was Windows, mine is on Linux.

These are my connections, each of them has been connected to multiple times, and each is set to be a favourite on the menu there.

![image](https://github.com/user-attachments/assets/2f52e6f6-e1fb-454f-98bb-1db34d4800f3)

But the favourites sections has no favourites.

![image](https://github.com/user-attachments/assets/192db570-ad96-4699-bf8b-c15e3805c6d9)

And recent sessions has no recent sessions. I had the Windows one open just now.

![image](https://github.com/user-attachments/assets/601bbcae-3443-4628-a651-fafcb8b10836)

### How to Reproduce

Install, configure, and connect to things.

### Expected Behavior

Recent sessions and favourites show up in their respective sections.

### Operating system(s) on local (controlling) side and remote (controlled) side

Linux -> Linux/Windows/Android

### RustDesk Version(s) on local (controlling) side and remote (controlled) side

1.3.3 on Manjaro Linux, 1.3.3 on Android, 1.3.5 on Windows, all connecting to each other.

### Screenshots

In bug description.

### Additional Context

The Windows and Android clients keep their history and favourites correctly.

**Comment by rustdesk** [comment]
@21pages he might have decode issue of peers. please follow it up. But i do not see any special characters in his peers. It might also because no permission to write to disk, or anything unknown.

**Reply by rustdesk** [reply]
# Please provide below info.
- [ ] please run `ls -ls ~/.config/rustdesk/peers/` and paste the log
- [ ] please paste the log in ~/.local/share/logs/RustDesk/rustdesk_rCURRENT.log
- [ ] run rustdesk from command and paste the output

**Comment by rustdesk** [comment]
We are quite busy now, you might wait for some days for @21pages  addressing this.

**Comment by nevdelap** [comment]
Not a rush, just letting you know.

```
18:35:27 zen ~ 2 ls -ls ~/.config/rustdesk/peers/
total 12
4 -rw------- 1 nevd nevd 1555 Dec  6 18:34 129230580.toml
4 -rw------- 1 nevd nevd 1608 Dec  5 16:14 1849570622.toml
4 -rw------- 1 nevd nevd 1599 Dec  1 14:07 94670260.toml
18:35:31 zen ~
18:35:31 zen ~ rustdesk
Cannot load libcuda.so.1
Cannot load libcuda.so.1
flutter: launch args: []
flutter: initializing FFI main
flutter: Failed to get documents directory: MissingPlatformDirectoryException(Unable to get application documents directory)
flutter: _appType:main,info1-id:7aa69f1a659d406d9f86f0a340c288b1,info2-name:Manjaro Linux,dir:
flutter: _globalFFI init
flutter: registerEventHandler callback_query_onlines address book peer
flutter: registerEventHandler load_address_book_peers address book peer
flutter: registerEventHandler callback_query_onlines group peer
flutter: registerEventHandler load_group_peers group peer
flutter: registerEventHandler callback_query_onlines recent peer
flutter: registerEventHandler load_recent_peers recent peer
flutter: registerEventHandler callback_query_onlines fav peer
flutter: registerEventHandler load_fav_peers fav peer
flutter: registerEventHandler callback_query_onlines discovered peer
flutter: registerEventHandler load_lan_peers discovered peer
flutter: _globalFFI init end
flutter: registerEventHandler native_ui native_ui
flutter: registerEventHandler check_software_update_finish check_software_update_finish
flutter: restore lpos: 1920.0/1035.0, offset:0.0/382.0, isMaximized: false, isFullscreen: false
flutter: handled by uni links: false
flutter: [MultiWindowHandler] active window changed: {0}
flutter: Saving frame: null: 1920.0/1035.0, offset:0.0/382.0, isMaximized:true, isFullscreen:false
/usr/bin/systemctl
gtk_main_quit: assertion 'main_loops != NULL' failed
flutter: launch args: [--no-server]
flutter: initializing FFI main
flutter: Failed to get documents directory: MissingPlatformDirectoryException(Unable to get application documents directory)
flutter: _appType:main,info1-id:7aa69f1a659d406d9f86f0a340c288b1,info2-name:Manjaro Linux,dir:
flutter: _globalFFI init
flutter: registerEventHandler callback_query_onlines address book peer
flutter: registerEventHandler load_address_book_peers address book peer
flutter: registerEventHandler callback_query_onlines group peer
flutter: registerEventHandler load_group_peers group peer
flutter: registerEventHandler callback_query_onlines recent peer
flutter: registerEventHandler load_recent_peers recent peer
flutter: registerEventHandler callback_query_onlines fav peer
flutter: registerEventHandler load_fav_peers fav peer
flutter: registerEventHandler callback_query_onlines discovered peer
flutter: registerEventHandler load_lan_peers discovered peer
flutter: _globalFFI init end
flutter: registerEventHandler native_ui native_ui
flutter: registerEventHandler check_software_update_finish check_software_update_finish
flutter: restore lpos: 1920.0/1035.0, offset:0.0/382.0, isMaximized: true, isFullscreen: false
flutter: handled by uni links: false
flutter: [MultiWindowHandler] active window changed: {0}
```

[rustdesk_rCURRENT.log](https://github.com/user-attachments/files/18035123/rustdesk_rCURRENT.log)

**Reply by 21pages** [reply]
@nevdelap Can you run this program and display logs containing the "DEBUG" string in the terminal? Please switch between recent tab and favorite tab.

https://github.com/21pages/rustdesk/releases/download/revert_linux_use_cpal_build/rustdesk-1.3.5-x86_64-peer-view.AppImage

**Comment by alpttex19** [comment]
no history not favorites on windows 11, rustdesk 1.3.9

**Comment by ElonJTrump** [comment]
I have the same problem with Windows 11 client. Try to uninstall and remove roaming data then re install again then connect to another PC but no recent session shown.

**Comment by mirostauder** [comment]
I have run into this issue on my new Debian 13 install,
please make sure to install the `xdg-user-dirs`package.
see: https://github.com/flutter/gallery/issues/979
@rustdesk - I would recommend to make it a package dependency

**Reply by markusantonuis** [reply]
This solved it for me on my armbian 26.2.1 (xfce) system. Thanks!

**Comment by Zixim** [comment]
`sudo apt install xdg-user-dirs` does indeed solve this.
